### ISSUE TITLE
mAP50, mAP50-95 on testing dataset

### ISSUE THREAD

**Issue by charlie10brown** [ultralytics/ultralytics#7991]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello,

I an using Yolov8 for instance segmentation on custom dataset and I want to acquire the mAP50, mAP50-95 after executing testing to include on the paper I am writing. 
I use this command `yolo task=segment mode=predict model={HOME}/runs/segment/train/weights/best.pt conf=0.25 source={dataset.location}/test/images save=true` but it returns annotated images and not a quantitative result, such as the evaluation metrics. 

Could you please help me with that?

Thanks a lot in advance.

**Comment by charlie10brown** [comment]
Thanks but after reading the official documentation, I still can't figure how to pass the test dataset `yolo detect val model=path/to/best.pt  # val custom model` on this command.

**Comment by glenn-jocher** [comment]
@charlie10brown hello,

To evaluate your model on a test dataset and obtain mAP50 and mAP50-95 metrics, you should use the `val` mode instead of `predict`. The command structure will look like this:

```
yolo task=segment mode=val model=path/to/best.pt data=path/to/data.yaml
```

Make sure your `data.yaml` file includes the correct paths to your test dataset. After running this command, the output will provide you with the mAP50 and mAP50-95 metrics for your model's performance on the test set.

Happy testing, and best of luck with your paper! 📊🍀

**Comment by charlie10brown** [comment]
Hello Glenn,

Thanks so much for your response :)

I got the testing results :)

**Comment by glenn-jocher** [comment]
@charlie10brown You're welcome! I'm glad to hear you got the results you needed. If you have any more questions or need further assistance, feel free to reach out. Happy researching! 🚀😊

**Comment by ayushraj2349** [comment]
The test path should be put in val argument or test itself?

config = {
   "train": "/content/drive/My Drive/helmet_dataset_YOLO/images/train",
   "val": "/content/drive/My Drive/helmet_dataset_YOLO/images/val",
   "test": "/content/drive/My Drive/helmet_dataset_YOLO/images/test"
   "nc": 3,
   "names": ['helmet','head','person']
}
it is correct or i need to pass test path to val argument for getting the results?

**Comment by glenn-jocher** [comment]
@ayushraj2349 to validate on a specific split you pass the `split` argument, i.e. `split="test"`. See https://docs.ultralytics.com/modes/val/#arguments

**Comment by ayushraj2349** [comment]
@glenn-jocher  Sir, after using this split="test" it will evaluate mAP on test images by going to the test directory specified by "test" argument in .yaml file , am i right just to confirm my understanidng?

and sir one more question, the mAP values printed in the results.csv file in the train folder, on which images they were calculated? were they of val or train parts?
and i have 3 classes in my dataset, why there is only 1 column for each, precision & recall in the results.csv file??

Again, thanks alot sir for responding to my queries.
![Screenshot (276)](https://github.com/ultralytics/ultralytics/assets/97942801/f38ccc9b-03d1-4e98-88e8-993a54f6705c)

**Comment by glenn-jocher** [comment]
@ayushraj2349 yes that's correct. Validation by default is always done on the validation set.

**Comment by ayushraj2349** [comment]
@glenn-jocher thansk u sir for the response. but i guess i have misunderstood my question. i want to ask why precision recall are only one , i mean there are 3 classes. which class does these values denote to?

and sir i want to store the result of mAP@0.50 in a variable or file, how to do that?

**Comment by glenn-jocher** [comment]
@ayushraj2349 hello! 👋

To obtain quantitative evaluation metrics like mAP50 and mAP50-95 on your testing dataset, you should use the `val` mode instead of `predict`. The `predict` mode is used for making predictions and visualizing them on images or videos, while `val` mode is designed for model evaluation and will provide you with the metrics you're looking for.

Here's the command you can use:

```bash
yolo task=segment mode=val model={HOME}/runs/segment/train/weights/best.pt data={your_data_yaml}
```

Make sure to replace `{your_data_yaml}` with the path to your dataset's YAML file that includes the test set paths. This command will run the validation process on the specified dataset and output the mAP50 and mAP50-95 metrics.

Good luck with your paper! 📄✨
